src/rules/function-max-empty-lines/index.js

```diff
@@ -0,0 +1,46 @@
+import { isNumber, repeat } from "lodash"
+import {
+  report,
+  ruleMessages,
+  styleSearch,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "function-max-empty-lines"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: "Unexpected empty line within function",
+})
+
+export default function (max) {
+  const maxAdjacentNewlines = max + 1
+
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: max,
+      possible: isNumber,
+    })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      if (decl.value.indexOf("(") === -1) { return }
+
+      const declString = decl.toString()
+
+      styleSearch({ source: declString, target: "\n", withinFunctionalNotation: true }, match => {
+        if (
+          declString.substr(match.startIndex + 1, maxAdjacentNewlines) === repeat("\n", maxAdjacentNewlines)
```

Comment:
Well, could be improved there as well :) Always learning.

It's probably trivial, but because stylelint can run through so much code I think it's worth taking little measures to reduce computation time. And theoretically at least I think that moving this outside the loop should do that: it just gets computed once instead of every time a `\n` is found.